certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/validation/model/ElementValidationDiagnosis.java

```diff
@@ -95,31 +95,53 @@ private List<ValidationError> invalidDiagnosisCodeValidationErrors(ElementData d
   }
 
   private List<ValidationError> incorrectOrderValidationErrors(ElementData data,
-      Optional<ElementId> categoryId, List<FieldId> elementDiagnoseIds) {
+      Optional<ElementId> categoryId, Map<FieldId, ElementValueDiagnosis> elementDiagnosesMap) {
     final var validationErrors = new ArrayList<ValidationError>();
     for (int index = order.size() - 1; index > 0; index--) {
-      if (!elementDiagnoseIds.contains(order.get(index))) {
+      if (!elementDiagnosesMap.containsKey(order.get(index))) {
         continue;
       }
 
-      while (index > 0 && missingDiagnosisValue(elementDiagnoseIds, index)) {
+      while (index > 0 && missingDiagnosisValue(elementDiagnosesMap, index)) {
         index--;
         final var orderId = order.get(index);
-        validationErrors.add(
-            errorMessage(data, orderId, categoryId, "Ange diagnoskod.")
-        );
-        validationErrors.add(
-            errorMessage(data, new FieldId(orderId.value() + DIAGNOSIS_DESCRIPTION), categoryId,
-                "Ange diagnosbeskrivning.")
-        );
-
+        if (missingDiagnosisCode(elementDiagnosesMap, orderId)) {
+          validationErrors.add(
+              errorMessage(data, orderId, categoryId, "Ange diagnoskod.")
+          );
+        }
+
+        if (missingDiagnosisDescription(elementDiagnosesMap, orderId)) {
+          validationErrors.add(
+              errorMessage(data, new FieldId(orderId.value()), categoryId,
+                  "Ange diagnosbeskrivning.")
+          );
+        }
       }
     }
     return validationErrors;
   }
 
-  private boolean missingDiagnosisValue(List<FieldId> elementDiagnoseIds, int index) {
-    return !elementDiagnoseIds.contains(order.get(index - 1));
+  private static boolean missingDiagnosisDescription(
+      Map<FieldId, ElementValueDiagnosis> elementDiagnosesMap, FieldId orderId) {
+    return elementDiagnosesMap.get(orderId) == null
+        || elementDiagnosesMap.get(orderId).description() == null
```

Comment:
This can be a function instead of duplicating the check between code and diagnosis since both are string

Reply:
Added method missingField to check the code and description value.